functions/src/index.test.ts

```diff
@@ -747,6 +856,20 @@ describe("token-service app", () => {
       expect(await getResourcesCount()).toEqual(1);
     });
 
+    it("fails if user is only a class/context member", async () => {
+      await createS3Settings();
+      const resource = await createS3Resource({
+        accessRules: [{ type: "context", contextId: user.contextId, platformId: user.platformId } as ContextAccessRule]
```

Comment:
Is there a test where the accessRules include both a user rule and a context rule?
This would be a common case where a resource has both, and a student matches both. 
And we'd want to make sure the student can still get credentials and update the resource.

Reply:
I've added this line in my last commit:
https://github.com/concord-consortium/token-service/blob/2d300ecaefb54d0a65a5b80bf8d049a519060c3e/functions/src/index.test.ts#L84

That way most of the tests that check owner rule will also deal with the member rule being in the same array.

Also, I've updated other tests that focus on the context rule to add a dummy user rule with `userId` pointing to another user. So, more real-life scenarios are being tested there.